### ISSUE TITLE
Support other languages in the landing page

### ISSUE THREAD

**Issue by ann26** [kartoza/IGRAC-GGIS#52]
Add support for other languages (French, Portuguese, etc) on the landing page contents

**Comment by meomancer** [comment]
hi @ArnaudIGRAC @clau1313 @timlinux 
The PR for translation is here:
https://github.com/kartoza/IGRAC-GGIS/pull/376 : this is for igrac
https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/pull/78 : this is for gwml2

To translate it, just go to 
IGRAC : https://github.com/kartoza/IGRAC-GGIS/tree/translation/django_project/igrac/locale
GWML : https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/tree/translation/locale

like below
![Selection_090](https://user-images.githubusercontent.com/4530905/107744715-c4642300-6d45-11eb-9b3c-55d21f7204c7.png)

and go to the language that needs to be translated
![Selection_091](https://user-images.githubusercontent.com/4530905/107744745-d1811200-6d45-11eb-8caf-542186da9368.png)

And use **.po** file
![Selection_092](https://user-images.githubusercontent.com/4530905/107744777-dfcf2e00-6d45-11eb-8707-cd3778dbcb3f.png)

example:
```
#: igrac/templates/base.html:319 igrac/templates/layers/layer_detail.html:288
msgid "Data"
msgstr ""
```

msgid is the english
and we just need to fill msgtr in the language
and don't forget to use ""

You can check what i did in indonesian
https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/blob/translation/locale/id/LC_MESSAGES/django.po

![Selection_093](https://user-images.githubusercontent.com/4530905/107744999-36d50300-6d46-11eb-9de9-8777c8b9a581.png)
![Selection_094](https://user-images.githubusercontent.com/4530905/107745009-39375d00-6d46-11eb-8855-2cdd4021037c.png)
![Selection_095](https://user-images.githubusercontent.com/4530905/107745020-3b012080-6d46-11eb-8145-7c9faeadba25.png)
![Selection_096](https://user-images.githubusercontent.com/4530905/107745032-3ccae400-6d46-11eb-8c7e-797ef0fad7c8.png)
![Selection_097](https://user-images.githubusercontent.com/4530905/107745035-3e94a780-6d46-11eb-9559-ae05f919f232.png)

Note:
CC @timlinux 
I wanted to use transifex, but i don't have account in there yet

**Comment by timlinux** [comment]
Ok @meomancer let's just use a texteditor for translation work for now - if the project grows a lot and we need many translators we can relook at transifex.

**Comment by timlinux** [comment]
Please merge so the translation work can take place

**Comment by clau1313** [comment]
Hi @meomancer, thanks. I guess this is Spanish:
![image](https://user-images.githubusercontent.com/36033176/108374917-c1fc4000-7201-11eb-815c-270d515b03a5.png)

If we change something here, will it appear automatically implemented in the GGIS? if not, do we have a deadline to make the translations? 

We will see the different languages when selecting an option here, right?
![image](https://user-images.githubusercontent.com/36033176/108375489-4b137700-7202-11eb-8e21-ed917c14daa8.png)

Sorry if my questions are a bit dumb, this is the first time I see something like this...

**Comment by meomancer** [comment]
> Hi @meomancer, thanks. I guess this is Spanish:
yes, that is correct @clau1313 

please update these file:
https://github.com/kartoza/IGRAC-GGIS/blob/translation/django_project/igrac/locale/es/LC_MESSAGES/django.po
https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/blob/translation/locale/es/LC_MESSAGES/django.po
Update the msgstr based on msgid above it

example:
```
msgid "GeoNode Search"
msgstr ""
```
fill msgstr "xxxx" with translation of GeoNode Search in spanish


To check the other language code, we could check it in here:
https://github.com/kartoza/geonode/blob/geonode.3.2-fix/geonode/settings.py#L189


> We will see the different languages when selecting an option here, right?
yes, you will see the different language

>If we change something here, will it appear automatically implemented in the GGIS? if not, do we have a deadline to make the translations?
Unfortunately no, you could provide me the text and i need to compile it to the project.
After that we can release it to the production (but before that we can test it on staging)

**Comment by clau1313** [comment]
Hi @meomancer, what is the text to be translated here?
![image](https://user-images.githubusercontent.com/36033176/108822226-136c4c80-75bf-11eb-8270-7cbcb0e4093b.png)

**Comment by clau1313** [comment]
Is this text correct?
![image](https://user-images.githubusercontent.com/36033176/108824795-6e537300-75c2-11eb-87b5-5a0e363b6069.png)

**Comment by clau1313** [comment]
It's me again! I found a similar case:
![image](https://user-images.githubusercontent.com/36033176/108825814-c474e600-75c3-11eb-8ff2-3ed67a028400.png)
shall I add the translation is separate lines after line 456?

**Comment by dimasciput** [comment]
Hi @clau1313 I think you can follow this example https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/blob/translation/locale/id/LC_MESSAGES/django.po

**Comment by meomancer** [comment]
thank you @dimasciput 
yes @clau1313 
please check the link from @dimasciput 
https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/blob/translation/locale/id/LC_MESSAGES/django.po
if it is empty, so we can skip to translate it

**Comment by clau1313** [comment]
Done, please check

**Comment by meomancer** [comment]
thank you @clau1313 

have you checked this one too?
https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/tree/translation
i don't see any commits from you

please check it with my example : https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/blob/translation/locale/id/LC_MESSAGES/django.po

**Comment by clau1313** [comment]
Hi @meomancer, I updated this two files:
https://github.com/kartoza/IGRAC-GGIS/blob/translation/django_project/igrac/locale/es/LC_MESSAGES/django.po
https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/blob/translation/locale/es/LC_MESSAGES/django.po

For the second file I couldn't update it directly, but it said that my suggested changes were going to be sent to the developers. Did you get them?

**Comment by meomancer** [comment]
hi @clau1313 
no, i don't get any of it
could you provide the file in here maybe?

**Comment by clau1313** [comment]
I saved it directly in the browser. When I try to edit it again I get the same message as yesterday:

![image](https://user-images.githubusercontent.com/36033176/109035985-721ded00-76c9-11eb-856c-b8f29c465f92.png)

maybe it is saved somewhere else?

**Comment by clau1313** [comment]
After I added my changes I clicked in "propose changes"

**Comment by meomancer** [comment]
hi @timlinux @ArnaudIGRAC 
could you give write access to @clau1313 for this repo?
https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase

**Comment by clau1313** [comment]
hi @timlinux and @ArnaudIGRAC, can you please also give access to @Alexia-Water to both files? (French versions)

@meomancer can you please check one more time if you got my proposed changes to the file? Otherwise I will have to translate the whole thing again...

**Comment by meomancer** [comment]
hi @clau1313 
i don't see any proposed changes
i think because i'm not admin

**Comment by ArnaudIGRAC** [comment]
hello Irwan, I think Alexia and Claudia have finalized the translation in French and Spanish. Can you please implement it?

**Comment by clau1313** [comment]
@ArnaudIGRAC I don't know if mine is ready because my proposed changes apparently dissapeared. I asked @meomancer but he said he cannot see them because he is not the admin. @timlinux did you perhaps receive my proposed changes for the translation of https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/blob/translation/locale/es/LC_MESSAGES/django.po ? otherwise I will have to translate the file again (which I don't prefer)

**Comment by ArnaudIGRAC** [comment]
@timlinux, as admin, could you please find out where the Spanish translation of Claudia has ended up (cfr messages above)? thanks

**Comment by meomancer** [comment]
hi @timlinux @ArnaudIGRAC @clau1313 
i think there is no solution for this, could we re translate it? thank you

**Comment by clau1313** [comment]
oh too bad, ok I will do it...

**Comment by meomancer** [comment]
Hi @clau1313 @ArnaudIGRAC 
could you check the staging for translation? i already deploy it
if it is ok, i will release it on production

**Comment by clau1313** [comment]
@meomancer I am not done with the translation yet, please wait a bit

**Comment by clau1313** [comment]
@meomancer ok, I am done with the translation

**Comment by meomancer** [comment]
hi @clau1313 
i have merged and deployed it to staging
could you create new branch and push it so i can pull it?
thank you

**Comment by meomancer** [comment]
ah, ok, i found it
please wait @clau1313

**Comment by meomancer** [comment]
It is done, thank you @clau1313 
could you check it on https://staging.igrac.kartoza.com/

CC @ArnaudIGRAC 

if it is ok, i will release it to production

**Comment by ArnaudIGRAC** [comment]
Hey Irwan, I would like to revise the French translation before you put it in production. Where can I do that?

**Comment by meomancer** [comment]
hi @ArnaudIGRAC 
You can change the french in here:
https://github.com/kartoza/IGRAC-GGIS/blob/develop/django_project/igrac/locale/fr/LC_MESSAGES/django.po
https://github.com/kartoza/IGRAC-WellAndMonitoringDatabase/blob/1983ea28a9e73c0f605df9cf83f04e04a581aada/locale/fr/LC_MESSAGES/django.po

Thank you

**Comment by ArnaudIGRAC** [comment]
Hello Irwan,

I have slightly revised the translation in French
[Doc 1 revised.txt](https://github.com/kartoza/IGRAC-GGIS/files/6295446/Doc.1.revised.txt)
[Doc 2 revised.txt](https://github.com/kartoza/IGRAC-GGIS/files/6295447/Doc.2.revised.txt)

**Comment by meomancer** [comment]
hi @ArnaudIGRAC 
I already deployed the revised translation on staging
Please check it https://staging.igrac.kartoza.com/

if it is ok, please close this and will be released on 1.1.7
https://github.com/kartoza/IGRAC-GGIS/issues/414
